### ISSUE TITLE
Some Audio issues

### ISSUE THREAD

**Issue by feliperce** [korlibs/korge#366]
Audio are not playing on JS.
Some times audio do not play on Android and some times play only the first 2, 3 seconds, never all the audio.

Audio are playing well on JVM and Native, but I can't stop the audio, If I call stop all my game freeze.

My game: https://github.com/feliperce/MolesAttack

on GameScene.kt
```
...

gameMusic = resourcesVfs["sound/msc_game.wav"].readSound()
val gameMusicChannel = gameMusic.play(infinitePlaybackTimes)

...

addUpdater {
            if (life <= 0) {

                gameTimer.close()

                val gameOver = GameOver(
                    score, minutes, seconds
                )

                launchImmediately {
                    gameMusicChannel.stop()
                    sceneContainer.changeTo<GameOverScene>(gameOver)
                }
            }
        }
```

**Comment by feliperce** [comment]
UPDATE:

The freezing issue are on infinitePlaybackTimes
If I remove the infinitePlaybackTimes my game stop freezing and stop the sound.

infinitePlayBackTimes and playForever are freezing when force stop.

And the JS: I searched and tested a lot, can't find the issue

`val gameMusicChannel = gameMusic.play(infinitePlaybackTimes)`

**Comment by soywiz** [comment]
Let's check a few things first:
* In the project you have there you are using KorGE `2.0.7.1` while the last published version is `2.0.9.1` ( https://github.com/korlibs/korge-hello-world/blob/cbfd6bff7cf8026f462574d0c016bbb0762a2a8c/gradle.properties#L1 )
* Additionally, there is korge-next where all the work is being done: https://github.com/korlibs/korge-next you can try to download that project then execute `./gradlew publishToMavenLocal` and use `korgePluginVersion=2.0.0.999` to use it.

Does your issue reproduce with those versions?

**Comment by feliperce** [comment]
Same results on 2.0.9.1 and 2.0.0.999:

- Now JS are Working great
- My game keep freezing on JVM and native
- Now Android are playing the sounds, but now my screen is black and after some times the sound freeze

**Comment by soywiz** [comment]
Reproduced. I will try to fix it before 2.1.0

**Comment by soywiz** [comment]
Actually, play is using the coroutineContext of the suspend function if not provided one. sceneInit cancels its coroutine context at the end. You can use this line instead:

```
val menuMusic = resourcesVfs["sound/msc_menu.wav"].readMusic().play(views.coroutineContext)
```
